### PR TITLE
Allow any trim type to be specified in initialization file

### PR DESCRIPTION
**Pull request by seanmcleod**
Currently the `<trim>` element in an initialization file can only be used to specify `0 - no trim` or `1 - ground trim`. This pull requests adds support for all the currently supported trim types to be specified and maintains backwards compatibility.

`- trim (0 for no trim, 1 for ground trim, 'Longitudinal', 'Full', 'Ground', 'Pullup', 'Custom', 'Turn') `

### CODE REVIEW

File: src/initialization/FGInitialCondition.cpp [hunk 1]
```diff
@@ -979,6 +980,27 @@ bool FGInitialCondition::LoadLatitude(Element* position_el)
 
 //******************************************************************************
 
+void FGInitialCondition::SetTrimRequest(std::string& trim)
```
Comment (bcoconni) [inline src/initialization/FGInitialCondition.cpp]: I think you need to use `const` for this code to compile under Travis CI
```c++
void FGInitialCondition::SetTrimRequest(const std::string& trim)
```

File: src/JSBSim.cpp [hunk 2]
```diff
@@ -454,10 +454,20 @@ int real_main(int argc, char* argv[])
...
-  if (FDMExec->GetIC()->NeedTrim()) {
-    trimmer = new JSBSim::FGTrim( FDMExec );
+  // Perform trim if requested via the initialization file
+  JSBSim::TrimMode icTrimRequested = (JSBSim::TrimMode)FDMExec->GetIC()->TrimRequested();
+  if (icTrimRequested != JSBSim::TrimMode::tNone) {
+    trimmer = new JSBSim::FGTrim( FDMExec, icTrimRequested );
     try {
+      // If PullUp requested then we need to pass in target Nlf
+      if (icTrimRequested == JSBSim::TrimMode::tPullup)
+        trimmer->SetTargetNlf(FDMExec->GetIC()->GetTargetNlfIC());
```
Comment (bcoconni) [inline src/JSBSim.cpp]: Why isn't that code in lines 462-464 included in `FGTrim::FGTrim` ?
I am pretty sure scripts will ignore the Nlf target otherwise.

**Comment by seanmcleod** [comment]
I've fixed the gcc compiler error regarding the `string` argument and fixed the merge conflict. But I see now via Travis that there is another gcc only (compiles fine on AppVeyor) error now regarding a goto and local variable initialization etc. which I'll look into later today.

I'll also respond later to the question regarding `targetNlf`.

File: src/JSBSim.cpp (hunk 2 above)
Comment (seanmcleod) [reply src/JSBSim.cpp]: I've moved the code to `FGTrim::FGTrim` which means the following will now also work which never worked previously in addition to pullup trim working from a `<trim>` element in the initialization file.

```xml
<!-- Pullup trim -->
<set name="simulation/do_simple_trim" value="4" />
```

**Pull request merged**